### PR TITLE
Add encryption params to rds instance resource

### PR DESCRIPTION
**Pull request by jarlef**
* Add storage_encrypted param which is default set to true
* Add kms_key_id option for scenarios using cross region replicas

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -117,3 +117,13 @@ variable "monitoring_role_arn" {
...
 }
+
+variable "storage_encrypted" {
+  description = "Specifies whether the DB instance is encrypted."
+  default     = true
+}
+
+variable "kms_key_id" {
+  description = "Specifies whether the DB instance is encrypted."
+  default     = "The ARN for the KMS encryption key. If creating an encrypted replica, set this to the destination KMS ARN."
```
Comment (itsdalmo) [inline variables.tf]: This default value seems like a mistake 😄

**Review by itsdalmo — COMMENTED** [review]
Looks good - and although I agree that encrypting the storage is a best practice, I don't think we can do it here since the user has to pass in a key ARN. So, I think `storage_encrypted` should default to `false` instead.

File: variables.tf [hunk 2]
```diff
@@ -117,3 +117,13 @@ variable "monitoring_role_arn" {
   description = "The ARN for the IAM role that permits RDS to send Enhanced Monitoring metrics to CloudWatch Logs"
   default     = ""
 }
+
+variable "storage_encrypted" {
+  description = "Specifies whether the DB instance is encrypted."
+  default     = true
```
Comment (colincoleman) [inline variables.tf]: Could you please change this to string "true" rather than using a boolean.  The reason for this is that future / older version may convert types differently but a string can always be compared with a string safely.

**Review by colincoleman — CHANGES_REQUESTED** [review]
A good improvement in principal - please see comment below

**Review by colincoleman — COMMENTED** [review]
Can you please also test this with storage_encrypted set to false and kms_key_id empty.  Some terraform resources fail when provided an empty optional key (example here https://github.com/terraform-providers/terraform-provider-aws/issues/5061)

I have checked the terraform aws provider and it looks like it is using their .GetOk method - so I don't think the above issue is a problem here.

File: variables.tf (hunk 1 above)
Comment (jarlef) [reply variables.tf]: Yes. I have fixed that. I dont think you have to provide a kms key

```
storage_encrypted - (Optional) Specifies whether the DB instance is encrypted. Note that if you are creating a cross-region read replica this field is ignored and you should instead declare kms_key_id with a valid ARN. The default is false if not specified.
```

**Review by colincoleman — APPROVED** [review]
